CSharp/ArticleProjects/PersistOutputs/OutputFilesExample.cs

```diff
@@ -0,0 +1,131 @@
+                virtualMachineSize: "standard_d1_v2",
+                targetDedicatedComputeNodes: nodeCount,
+                virtualMachineConfiguration: new VirtualMachineConfiguration(
+                    skuAndImage.Image,
+                    skuAndImage.Sku.Id));
+
+            // Commit the pool to the Batch service
+            GettingStartedCommon.CreatePoolIfNotExistAsync(batchClient, pool).Wait();
+
+            CloudJob job = batchClient.JobOperations.CreateJob(jobId, new PoolInformation { PoolId = poolId });
+
+            // Commit the job to the Batch service
+            job.Commit();
+            Console.WriteLine($"Created job {jobId}");
+
+            CloudBlobContainer container = storageAccount.CreateCloudBlobClient().GetContainerReference(containerName);
+            container.CreateIfNotExists();
+
+            string containerSas = container.GetSharedAccessSignature(new SharedAccessBlobPolicy()
+            {
+                Permissions = SharedAccessBlobPermissions.Write,
+                SharedAccessExpiryTime = DateTimeOffset.UtcNow.AddDays(1)
+            });
+            string containerUrl = container.Uri.AbsoluteUri + containerSas;
+
+            // Create a series of simple tasks which dump the task environment to a file and then write random values to a text file
+            IEnumerable<CloudTask> tasks = Enumerable.Range(1, 20).Select(i =>
+                {
+                    var taskId = i.ToString().PadLeft(3, '0');
+                    return new CloudTask(taskId, $"cmd /v:ON /c \"echo off && set && (FOR /L %i IN (1,1,100000) DO (ECHO !RANDOM!)) > output.txt\"")
+                    {
+                        OutputFiles = new List<OutputFile>
+                        {
+                            new OutputFile(
+                                filePattern: @"..\std*.txt",
+                                destination: new OutputFileDestination(new OutputFileBlobContainerDestination(
+                                    containerUrl: containerUrl,
+                                    path: taskId)),
+                                uploadOptions: new OutputFileUploadOptions(
+                                    uploadCondition: OutputFileUploadCondition.TaskCompletion)),
+                            new OutputFile(
+                                filePattern: @"output.txt",
+                                destination: new OutputFileDestination(new OutputFileBlobContainerDestination(
+                                    containerUrl: containerUrl,
+                                    path: taskId + @"\output.txt")),
+                                uploadOptions: new OutputFileUploadOptions(
+                                    uploadCondition: OutputFileUploadCondition.TaskCompletion)),
+                        }
+                    };
+                }
+            );
+
+            batchClient.JobOperations.AddTask(jobId, tasks);
+            job.Refresh();
+            tasks = job.ListTasks();
```

Comment:
The other code does a Refresh (or possibly a GetJob, maybe Refresh didn't exist when I wrote it), followed by `job.AddTask`.  Let's try to minimise the differences between the parts of each routine that _don't_ deal with task outputs - otherwise the reader is going to be asking 'why do add tasks one way for conventions and a different way for file outputs?'